### DISCUSSION TITLE
[Bug]: Group not rendering properly on zoom

### DISCUSSION THREAD

**Discussion by lh-sn** [fabricjs/fabric.js#9168] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta11

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

None

### Link To Reproduction

https://codesandbox.io/p/sandbox/github/lh-sn/fabric-demo/tree/main

### Steps To Reproduce

1. Increase and decrease zoom

https://github.com/fabricjs/fabric.js/assets/137523466/7d88334e-b326-4514-ade1-600bd3bd89fc

### Expected Behavior

- It should draw full object
- bounding box should not include outline path

### Actual Behavior

For zoom level under 0.5, it draws with no issues, but, for greater values, it draws randomly.

### Error Message & Stack Trace

```bash
---
```

**Comment by lh-sn** [comment]
@ShaMan123, do you know what I am lacking? I would greatly appreciate it if you could offer some insight here.

**Comment by lh-sn** [comment, marked as answer]
@ShaMan123 Thanks for adding some updates related to this.  Bounding box with padding and zoom != 1 doenst work as expected.

Demo: https://codesandbox.io/p/sandbox/admiring-bohr-3x7vq9?file=%2Fsrc%2Fillustration.ts%3A1%2C1

https://github.com/fabricjs/fabric.js/assets/137523466/4916da10-913f-41a7-ab81-0e393cca92ed

I am using a workaround for the second box, changing padding value before drawing borders

```ts
 drawBorders(
   ctx: CanvasRenderingContext2D,
    options: TQrDecomposeOut,
    styleOverride: TStyleOverride
  ): void {
    let size;
    const zoom = this.canvas.getZoom();
    this.padding = -(STROKE_WIDTH / 2) * zoom;
   ... rest of code
}
```

**Comment by ShaMan123** [comment]
Looks like you got it.
Regarding padding etc. We are thinking of creating a post render effect/hook.
That way it will not render to cache and will be easier to keep in sync and reason about.
Consider experimenting with it.

**Reply by ShaMan123** [reply]
And stay tuned to the redo coords PR that fixes most of the known problems related to coords

**Reply by StringKe** [reply]
I see that you contributed the need for more pr, thanks for your contribution and effort. 

Some of them are not merged yet and there is no release version, is there any way I can experience these features in advance?

**Reply by ShaMan123** [reply]
I wanted to release a candidate for each merged PR but that didn't happen.
Because the build process is straight forward in v6 you can use a branch, PR or a fork in npm
You only need to add a postinstall script
```bash
"postinstall": "cd node_modules/fabric && npm i --include=dev && npm run build -- -f"
```

You can also use `npm link` to develop fabric locally, refer to our `npm start` cmd
